pyfr/plugins/fluidforce.py

```diff
@@ -54,6 +59,18 @@ def __init__(self, intg, cfgsect, suffix):
             header = ['t', 'px', 'py', 'pz'][:self.ndims + 1]
             if self._viscous:
                 header += ['vx', 'vy', 'vz'][:self.ndims]
+            if self.morigin:
+                # Add headers of moments. Moments always have one and three components
+                # in 2D and 3D simulations, respectively
+                self._mcomponents = 3 if self.ndims == 3 else 1
+                if self.ndims == 2:
+                    header += ['mpz']
+                    if self._viscous:
+                        header += ['mvz']
+                else:
+                    header += ['mpx', 'mpy', 'mpz']
```

Comment:
Given the number of moments we should be able to play a similar trick to `['vx', 'vy', 'vz'][:self.ndims]` to simplify this block.

Reply:
This is tricky since for 2D cases it's only the z direction component which is non-zero which is different from the forces in which we have `ndims` non-zero components for both 2D and 3D cases.

Reply:
`['mpx', 'mpy', 'mpz'][3 - mcomp:]` I think will do it.

Reply:
Makes sense, thanks!